### PR TITLE
Expose system connection authentication in external packs info

### PR DESCRIPTION
**Pull request by kelsey-codaio**
Expose system connection authentication in external packs info

### CODE REVIEW

**Review by patrick-codaio — APPROVED** [review]
Wow, can't believe that wasn't there before, thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This changelog keeps track of all changes to the Packs SDK. We follow convention
 ### Added
 
 - `coda execute` now allows for executing metadata formulas and sync updates.
+- Add `systemConnectionAuthentication` to `ExternalPackVersionMetadata`.
```
Comment (jonathan-codaio) [inline CHANGELOG.md:10]: For internal-only stuff we needn't update the changelog, that's meant for public consumption.

**Review by jonathan-codaio — CHANGES_REQUESTED** [review]
Hmm what are we trying to accomplish here?

First off we can't just pass through systemConnectionAuthentication, this metadata object is meant to be a plain JSON object but the authentication definition has code in it (like getConnectionName), so anything like that needs to be transformed first.

I think the reason this wasn't included before was that end users aren't supposed to know or care about system authentication, so it doesn't need to be in external metadata.  Do end users need to know all the sudden?

**Comment by kelsey-codaio** [comment]
> Hmm what are we trying to accomplish here?
> 
> First off we can't just pass through systemConnectionAuthentication, this metadata object is meant to be a plain JSON object but the authentication definition has code in it (like getConnectionName), so anything like that needs to be transformed first.
> 
> I think the reason this wasn't included before was that end users aren't supposed to know or care about system authentication, so it doesn't need to be in external metadata. Do end users need to know all the sudden?

Yeah, make sense. I was trying to address [this issue](https://staging.coda.io/d/_dL5PQJ5pk51?encodedLocationContext=eyJjb21tZW50VGhyZWFkSWQiOiJyLXE2ZzItSkh6aTYiLCJwYWdlSWQiOiJzZWN0aW9uLW5STVVxVnlRZjgiLCJ0eXBlIjoidGFibGUiLCJvYmplY3RJZCI6InRhYmxlLXV1dTFnTDJqcmwiLCJncmlkSWQiOiJncmlkLUUweXpvR1E0M1UiLCJyb3dJZCI6ImktN1dCQjZJVFYwViIsInZpZXdNb2RlIjoiZnVsbCJ9), which required knowing whether or not a pack was using system authentication from `syncTableGrid`. Thoughts on how else we could accomplish this? @jonathan-codaio

**Comment by jonathan-codaio** [comment]
Hmm can we condition editability on

```typescript
pack.authentication &&
pack.authentication.type != ExternalPackAuthenticationType &&
syncFormula.connectionRequirement &&
syncFormula.connectionRequirement != ConnectionRequirement.None
```

I'm not sure that we need to know about system auth here. If `pack.authentication` is blank or None, it means there is no authentication that the end user needs to care about, so any connectionRequirement is either erroneous or an implementation detail and we can just ignore.

**Comment by kelsey-codaio** [comment]
> Hmm can we condition editability on
> 
> ```ts
> pack.authentication &&
> pack.authentication.type != ExternalPackAuthenticationType.None &&
> syncFormula.connectionRequirement &&
> syncFormula.connectionRequirement != ConnectionRequirement.None
> ```
> 
> I'm not sure that we need to know about system auth here. If `pack.authentication` is blank or None, it means there is no authentication that the end user needs to care about, so any connectionRequirement is either erroneous or an implementation detail and we can just ignore.

The problem we were running into was that [Eric Koleda said](https://staging.coda.io/d/_dL5PQJ5pk51?encodedLocationContext=eyJjb21tZW50VGhyZWFkSWQiOiJyLXE2ZzItSkh6aTYiLCJwYWdlSWQiOiJzZWN0aW9uLW5STVVxVnlRZjgiLCJ0eXBlIjoidGFibGUiLCJvYmplY3RJZCI6InRhYmxlLXV1dTFnTDJqcmwiLCJncmlkSWQiOiJncmlkLUUweXpvR1E0M1UiLCJyb3dJZCI6ImktN1dCQjZJVFYwViIsInZpZXdNb2RlIjoiZnVsbCJ9) it should work even if the user doesn't explicitly set `connectionRequirement`. (Right now, `syncTableGrid.connectionRequirement` is undefined when using the example pack code snippet in the bug.) @jonathan-codaio

**Pull request closed without merging**